Paymetheus/ViewModels/PurchaseTicketsViewModel.cs

```diff
@@ -0,0 +1,404 @@
+        {
+            var synchronizer = ViewModelLocator.SynchronizerViewModel as SynchronizerViewModel;
+            if (synchronizer != null)
+            {
+                SelectedAccount = synchronizer.Accounts[0];
+            }
+
+            FetchStakeDifficultyCommand = new DelegateCommand(FetchStakeDifficultyAction);
+            FetchStakeDifficultyCommand.Execute(null);
+
+            _purchaseTickets = new DelegateCommand(PurchaseTicketsAction);
+            _purchaseTickets.Executable = false;
+        }
+
+        private DelegateCommand _purchaseTickets;
+        public ICommand Execute => _purchaseTickets;
+
+        private bool _poolChecked = false;
+        public bool PoolChecked
+        {
+            get { return _poolChecked; }
+            set { _poolChecked = value; RaisePropertyChanged(); }
+        }
+
+        private bool _feesChecked = false;
+        public bool FeesChecked
+        {
+            get { return _feesChecked; }
+            set { _feesChecked = value; RaisePropertyChanged(); }
+        }
+
+        private AccountViewModel _selectedAccount;
+        public AccountViewModel SelectedAccount
+        {
+            get { return _selectedAccount; }
+            set { _selectedAccount = value; RaisePropertyChanged(); }
+        }
+
+        private Address _ticketAddress = null;
+        private string _ticketAddressString;
+        public string TicketAddressString
+        {
+            get { return _ticketAddress.ToString(); }
+            set {
+                _ticketAddressString = value;
+
+                _ticketAddress = null;
+                if (_ticketAddressString == "") {
+                    return;
+                }
+
+                try
+                {
+                    _ticketAddress = Address.Decode(_ticketAddressString);
+                }
+                finally
+                {
+                    _settingsAreValid();
+                }
+            }
+        }
+
+        private Address _poolAddress = null;
+        private string _poolAddressString;
+        public string PoolAddressString
+        {
+            get { return _poolAddress.ToString(); }
+            set
+            {
+                _poolAddressString = value;
+
+                _poolAddress = null;
+                if (_poolAddressString == "")
+                {
+                    return;
+                }
+
+                try
+                {
+                    _poolAddress = Address.Decode(_poolAddressString);
+                }
+                finally
+                {
+                    _settingsAreValid();
+                }
+            }
+        }
+
+        private double _poolFees = 0.0;
+        public double PoolFees
+        {
+            get { return _poolFees; }
+            set
+            {
+                var _testPoolFees = value * 100.0;
+                if (_testPoolFees != Math.Floor(_testPoolFees))
+                {
+                    _poolFees = 0.0;
```

Comment:
You can set _poolFees to 0 (representing an invalid value) exactly once like this:

```
try
{
    if (errorCondition1)
        throw new ArgumentException("error 1");
    if (errorCondition2)
        throw new ArgumentException("error 2");
    _poolFees = value;
}
catch
{
    _poolFees = 0.0;
    throw;
}
finally
{
    EnableOrDisableSendCommand();
}
```